bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/handler/IpCameraHandler.java

```diff
@@ -0,0 +1,2020 @@
+                                    if (msg instanceof HttpMessage) {
+                                        // very start of stream only
+                                        ReferenceCountUtil.retain(msg, 1);
+                                        firstStreamedMsg = msg;
+                                        streamToGroup(firstStreamedMsg, mjpegChannelGroup, true);
+                                    }
+                                }
+                            } else if (contentType.contains("image/jp")) {
+                                if (bytesToRecieve == 0) {
+                                    bytesToRecieve = 768000; // 0.768 Mbyte when no Content-Length is sent
+                                    logger.debug("Camera has no Content-Length header, we have to guess how much RAM.");
+                                }
+                                incomingJpeg = new byte[bytesToRecieve];
+                            }
+                            if (closeConnection) {
+                                lock.lock();
+                                try {
+                                    byte indexInLists = (byte) listOfChannels.indexOf(ctx.channel());
+                                    if (indexInLists >= 0) {
+                                        listOfChStatus.set(indexInLists, (byte) 0);
+                                    } else {
+                                        logger.debug("!!!! Could not find the ch for a Connection: close URL:{}",
+                                                requestUrl);
+                                    }
+                                } finally {
+                                    lock.unlock();
+                                }
+                            }
+                        }
+                    }
+                }
+                if (msg instanceof HttpContent) {
+                    if (mjpegUri.contains(requestUrl)) {
+                        // multiple MJPEG stream packets come back as this.
+                        ReferenceCountUtil.retain(msg, 1);
+                        streamToGroup(msg, mjpegChannelGroup, true);
+                    } else {
+                        HttpContent content = (HttpContent) msg;
+                        // Found some cameras uses Content-Type: image/jpg instead of image/jpeg
+                        if (contentType.contains("image/jp")) {
+                            for (int i = 0; i < content.content().capacity(); i++) {
+                                incomingJpeg[bytesAlreadyRecieved++] = content.content().getByte(i);
+                            }
+                            if (content instanceof LastHttpContent) {
+                                lockCurrentSnapshot.lock();
+                                currentSnapshot = incomingJpeg;
+                                lockCurrentSnapshot.unlock();
+                                processSnapshot();
```

Comment:
Why not just pass in the snapshot to be processed into `processSnapshot` as an argument?

Reply:
Thanks, that made things cleaner.